### PR TITLE
Spec updates

### PR DESCRIPTION
**Pull request by dmkash**
Starting over with a new spec suite!

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -102,7 +112,8 @@ end
 group :test do
   # Addressable Required by WebMock but breaks everything at 2.2.5
   # Can use latest addressable when pull request is accepted: https://github.com/sporkmonger/addressable/pull/33
-  gem 'shoulda'
+  gem 'shoulda', require: false
+  gem 'faker'
   gem 'addressable'
```
Comment (KBalderson) [inline Gemfile]: @dmkash would it make sense to also remove a lot of the testing gems? here are some that I think are a bit superfluous:

shoulda
addressable
linguistics
rack-test
simplecov
steak
timecop
webrat

I know you've talked about timecop, so feel no need to justify leaving it in. Thoughts?

**Comment by KBalderson** [comment]
@dmkash it looks like this isn't up to date with the most recent master. I don't mind merging it in via the command line, let me know how you'd like to proceed.

**Comment by dmkash** [comment]
I will merge it -- thank you for catching that!!

**Comment by dmkash** [comment]
@KBalderson I just updated with master.

**Comment by dmkash** [comment]
@KBalderson I will probably end up adding shoulda back in.  Do you not like that syntax?  Also, I was thinking about seeing if we can upgrade RSpec.

**Comment by KBalderson** [comment]
@dmkash go ahead and keep shoulda... I haven't used it before... so, that's why I thought it was unnecessary

it breaks specs if you don't include the "require: false" in the gemfile though.

**Comment by KBalderson** [comment]
@dmkash Were you going to remove any of the other gems in that list?

addressable
linguistics
rack-test
simplecov
steak
timecop
webrat

**Comment by dmkash** [comment]
@KBalderson Yeah, I think so.  Since we're starting over from scratch, we can just include the ones that we think are actually useful.

**Comment by dmkash** [comment]
@KBalderson I removed all of those superfluous gems and did some upgrading of RSpec.  We should be using the "expect" syntax.

**Comment by dmkash** [comment]
Also note that there is now a `spec/rspec_helper.rb` and `spec/spec_helper.rb` file.  Specs that require the Rails environment (99% of them) will require the `rspec_helper` file instead of the `spec_helper` file.

**Pull request merged**